spatialos-sdk/src/worker/vtable.rs

```diff
@@ -1,5 +1,96 @@
+use crate::worker::component::{Component, ComponentId};
+use crate::worker::schema::{
+    DataPointer, SchemaCommandRequest, SchemaCommandResponse, SchemaComponentData,
+    SchemaComponentUpdate,
+};
 use spatialos_sdk_sys::worker::*;
-use std::ptr;
+use std::collections::HashMap;
+use std::os::raw;
+use std::sync::Arc;
+use std::{mem, ptr};
+
+inventory::collect!(VTable);
+
+lazy_static::lazy_static! {
+    pub(crate) static ref DATABASE: ComponentDatabase = {
+
+        let mut raw_vtables = Vec::new();
+        let mut vtables = HashMap::new();
+
+        for table in inventory::iter::<VTable>.into_iter() {
+            raw_vtables.push(table.worker_vtable);
+            vtables.insert(table.component_id(), table.clone());
+        }
+
+        ComponentDatabase {
+            vtables,
+            raw_vtables
+        }
+    };
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct ComponentDatabase {
+    vtables: HashMap<ComponentId, VTable>,
+    raw_vtables: Vec<Worker_ComponentVtable>,
+}
+
+impl ComponentDatabase {
+    pub(crate) fn has_vtable(&self, id: ComponentId) -> bool {
+        self.vtables.contains_key(&id)
+    }
+
+    pub(crate) fn get_vtable(&self, id: ComponentId) -> Option<&VTable> {
+        self.vtables.get(&id)
+    }
+
+    pub(crate) fn to_worker_sdk(&self) -> *const Worker_ComponentVtable {
+        self.raw_vtables.as_ptr()
+    }
+
+    pub(crate) fn len(&self) -> usize {
+        self.vtables.len()
+    }
+}
+
+unsafe impl Sync for ComponentDatabase {}
+unsafe impl Send for ComponentDatabase {}
+
+#[derive(Clone, Debug)]
+pub struct VTable {
+    pub worker_vtable: Worker_ComponentVtable,
```

Comment:
Could this be `pub(crate)` instead of `pub`? It shouldn't need to be exposed publicly, as far as I can tell.

```suggestion
    pub(crate) worker_vtable: Worker_ComponentVtable,
```

Reply:
Done! :)